include/zephyr/net/ocpp.h

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) 2024 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file ocpp.h
+ *
+ * @defgroup ocpp_api OCPP library
+ * @ingroup networking
+ * @{
+ * @brief OCPP Charge Point Implementation
+ *
+ * @note The implementation assumes Websocket module is enabled.
+ *
+ * @note By default the implementation uses OCPP version 1.6.
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_OCPP_H
+#define ZEPHYR_INCLUDE_NET_OCPP_H
+
+#include <stdio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/net/net_ip.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define CISTR50	50
+
+/**
+ * @brief OCPP IdTage authorization status in result to ocpp request
+ *        authorization
+ */
+typedef enum {
+	AUTH_ACCEPTED,
+	AUTH_BLOCKED,
+	AUTH_EXPIRED,
+	AUTH_INVALID,
```

Comment:
If the values are not fixed, it would be nice if the AUTH_INVALID would be first so that static status value would get 0 (invalid) values as a default one.
Also you need to name space the enum values, just add `OCPP_...` prefix.

We usually only use typedefs for function pointers in Zephyr, so I would suggest you remove those in this file (unless it is a function pointer you are defining).